dds/DCPS/RTPS/ICE/EndpointManager.h

```diff
@@ -121,13 +120,13 @@ struct EndpointManager {
     guid_pair_to_checklist_.erase(pos);
   }
 
-  void set_responsible_checklist(const std::string& a_username, Checklist* a_checklist)
+  void set_responsible_checklist(const std::string& a_username, ChecklistPtr a_checklist)
   {
     OPENDDS_ASSERT(!username_to_checklist_.count(a_username));
     username_to_checklist_[a_username] = a_checklist;
   }
 
-  void unset_responsible_checklist(const std::string& a_username, Checklist* a_checklist)
+  void unset_responsible_checklist(const std::string& a_username, ChecklistPtr a_checklist)
```

Comment:
Should `a_checklist` be a plain pointer here?  It's only used for comparison -- this method can't retain a reference.  This is often useful to make a part of the method signature so that callers that are in a destructor (for example) don't have to deal with shared_from_this or otherwise go through an unneeded increment/decrement cycle.  (This probably applies to all of the "unset_*" methods)

Reply:
Probably but I would prefer to deal with that later.